File: Day 2/python/main.py
def shuffle_list(the_list):
    n = len(nums) // 2
    result = []
    
    for i in range(n):
        result.append(nums[i])
        result.append(nums[i + n])
    
    return result

nums = []
while (True):
    print("\nEnter number of elements the array will have. It must be at least 4, and an even number.")
    count = int(input("Enter here: "))
    if count >= 4 and count % 2 == 0:
        break
    else:
        print("\nInvalid.")

for i in range(1, count + 1):
    num = int(input(f"\nEnter integer {i}: "))
    nums.append(num)

print(shuffle_list(nums))

File: Day 2/kotlin/main.kt
import java.util.Scanner

fun shuffleList(theList: List<Int>): List<Int> {
    val n = theList.size / 2
    val ans = mutableListOf<Int>()
    for (i in 0..n - 1) {
        ans.add(theList[i])
        ans.add(theList[i + n])
    }
    return ans
}

fun main() {
    val nums = mutableListOf<Int>()

    val reader = Scanner(System.`in`)

    var count = 0;

    while (true) {
        println("Enter number of elements the array will have. It must be at least 4, and an even number.")
        count = reader.nextInt()
        if (count >= 4 && count % 2 == 0)
            break
        else
            println("Invalid.")
    }

    for (i in 1..count) {
        print("\nEnter integer $i: ")
        val num = reader.nextInt()
        nums.add(num)
    }

    println(shuffleList(nums))
}